src/main.rs

```diff
@@ -31,6 +31,18 @@ struct CliOpts {
     /// `WEIGHT`s are integers. If unspecifified, the default weight of 1000 is used.
     #[structopt(long = "overrides", env = "LINKERD2_MOCK_DST_OVERRIDES", default_value = "", parse(try_from_str = parse_overrides))]
     overrides: OverridesSpec,
+
+    /// A directory that is dynamically watched for endpoints updates
+    ///
+    /// The directory contains files with names in the form of {dst.name}:{port}. Each file should
+    /// contain the json representation of a list of `EndpointMeta` objects. Not that if such a
+    /// directory is provided the `endpoints` and `overrides` opts will be ignored and the discovery
+    /// state will be derived from the contents of the directory only.
+    #[structopt(
+        long = "endpoints_watch_directory",
+        env = "LINKERD2_MOCK_DST_ENDPOINTS_WATCH_DIRECTORY"
```

Comment:
If we have a value for `endpoints_watch_directory`, we will ignore the value of `overrides` and `endpoints`. We can get `structopt` to validate that these flags are not used together by using `clap`'s [`Arg::conflicts_with_all`](https://docs.rs/clap/2.33.1/clap/struct.Arg.html#method.conflicts_with_all) as a [`structopt` raw method](https://docs.rs/structopt/0.3.15/structopt/#raw-methods):
```suggestion
        env = "LINKERD2_MOCK_DST_ENDPOINTS_WATCH_DIRECTORY",
        conflicts_with_all = &["overrides", "endpoints"],
```